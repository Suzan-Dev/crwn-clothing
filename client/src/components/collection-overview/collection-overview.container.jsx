import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import WithSpinner from '../with-spinner/with-spinner.component';
import CollectionOverview from './collection-overview.component';

import { selectIsFetching } from '../../redux/shop/shop.selector';

const mapStateToProps = createStructuredSelector({
  isLoading: selectIsFetching,
});

const CollectionOverviewContainer = compose(
  connect(mapStateToProps),
  WithSpinner
)(CollectionOverview);

export default CollectionOverviewContainer;
