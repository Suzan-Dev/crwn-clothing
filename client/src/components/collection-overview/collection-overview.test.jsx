import React from 'react';
import { shallow } from '../../setupTests';
import { CollectionOverview } from './collection-overview.component';

describe('Testing: CollectionOverview component!', () => {
  const mockCollections = [{ id: 1 }, { id: 2 }];

  const mockProps = {
    collections: mockCollections,
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CollectionOverview {...mockProps} />);
  });

  test('should render CollectionOverview component.', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should render equal num of CollectionPreview component as in collections array.', () => {
    expect(wrapper.find('CollectionPreview').length).toBe(mockCollections.length);
  });
});
