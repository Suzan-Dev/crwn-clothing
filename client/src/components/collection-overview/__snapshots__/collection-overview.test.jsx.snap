// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing: CollectionOverview component! should render CollectionOverview component. 1`] = `
"<div className=\\"collection-overview\\">
  <CollectionPreview />
  <CollectionPreview />
</div>"
`;
